geode-tcp-server/src/main/java/org/apache/geode/distributed/internal/tcpserver/TcpClient.java

```diff
@@ -200,21 +200,28 @@ public Object requestToServer(HostAndPort addr, Object request, int timeout,
       out.flush();
 
       if (replyExpected) {
-        DataInputStream in = new DataInputStream(sock.getInputStream());
-        if (debugVersionMessage != null && logger.isDebugEnabled()) {
-          logger.debug(debugVersionMessage);
-        }
-        in = new VersionedDataInputStream(in, serverVersion);
+        DataInputStream in = null;
         try {
```

Comment:
Could this have used a `try-with-resources`?

Reply:
Good catch, thanks.